### PR TITLE
535 - Add IT for deploying artifacts

### PR DESCRIPTION
**Pull request by genryxy**
Part of #535 
Add IT for deploying release and snapshot artifacts. Also replace `StandardCharsets.UTF_8.displayName()` with `StandardCharsets.UTF_8` in `URLEncodedUtils.parse()` since this method was deprecated.

### CODE REVIEW

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy that's not quite what we need, tests in artipie should start whole artipie system, not individual adapter slice.

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy please check my comment

**Comment by genryxy** [comment]
@olenagerasimova thanks, finally fixed, please take a look

File: src/test/java/com/artipie/maven/MavenITCase.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+    private void addFilesToStorage(final String resource, final Key key)
+        throws InterruptedException {
+        final Storage resources = new FileStorage(
+            new TestResource(resource).asPath()
+        );
+        final BlockingStorage bsto = new BlockingStorage(resources);
+        bsto.list(Key.ROOT).stream()
+            .map(Key::string)
+            .forEach(
```
[the remark this answers is not in the document]
Comment (genryxy) [reply src/test/java/com/artipie/maven/MavenITCase.java]: @g4s8 thanks, simplified

File: src/test/java/com/artipie/maven/MavenITCase.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+            this.subdir.resolve(pom),
+            String.format(
+                Files.readString(this.subdir.resolve(pom)),
+                this.port
+            ).getBytes()
+        );
+    }
+
+    /**
+     * Executes dnf command in container.
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/maven/MavenITCase.java:224]: @genryxy javadoc seems inaccurate, I think we can remove it as it's not required in tests and other methods have no comments

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy just one comment, please, take a look

**Comment by genryxy** [comment]
@olenagerasimova thanks, removed comment

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#537` is not assigned, can't get performer

**Pull request merged**